aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,196 @@
+					 
+.. literalinclude:: validation/sample/ClassicMovieAttribute.cs
+   :language: c#
+   :lines: 11-37
+   :dedent: 4
+   
+The ``movie`` variable above represents a ``Movie`` object that contains the data from the form submission to validate. In this case, the validation code checks the date and genre in the ``IsValid`` method of the ``ClassicMovieAttribute`` class as per the rules. Upon successful validation ``IsValid`` returns a ``ValidationResult.Success`` code, and when validation fails, a ``ValidationResult`` with an error message. When a user modifies the ``Genre`` field and submits the form, the ``IsValid`` method of the ``ClassicMovieAttribute`` will verify whether the movie is a classic. Like any built-in attribute, apply the ``ClassicMovieAttribute`` to a property such as ``ReleaseDate`` to ensure validation happens, as shown in the previous code sample. 
+
+Alternatively, this same code could be placed in the model instead by implementing ``IValidatableObject`` which contains a ``Validate`` method. This works well for one-off validations like in the sample.
+
+ .. literalinclude:: validation/sample/MovieIValidatable.cs
+   :language: c#
+   :lines: 35-48
+   :dedent: 8
+  
+Client side validation
+----------------------
+Client side validation is a great convenience for users. It saves time they would otherwise spend waiting for a round trip to the server. In business terms, even a few seconds multiplied hundreds of times each day adds up to be a lot of time, expense, and frustration. Straightforward and immediate validation enables users to work more efficiently and produce better quality input and output. 
+
+You must have a view with the proper JavaScript script references in place for client side validation to work. 
+
+.. code-block:: html
+
+  <script src="~/lib/jquery/jquery.js"></script>
+  <script src="~/lib/jquery-validation/dist/jquery.validate.js"></script>
+  <script src="~/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.js"></script>
+
+MVC uses data annotations in addition to type metadata from model properties to validate data and display any error messages using JavaScript. When you use MVC to render form elements from a model using `Tag Helpers <https://docs.asp.net/en/latest/mvc/views/tag-helpers/index.html>`_ or `HTML helpers <https://docs.asp.net/en/latest/mvc/views/html-helpers.html>`_ it will add HTML 5 `data- attributes. <http://w3c.github.io/html/dom.html#embedding-custom-non-visible-data-with-the-data-attributes>`_ in the form elements that need validation, as shown below. MVC generates the ``data-`` attributes for both built-in and custom attributes. The ``data-val-required`` attribute below contains an error message to display if the user doesn't fill in the release date field, and that message displays in the accompanying ``<span>`` element. You can display validation errors on the client using the relevant tag helpers as shown here:
```

Comment:
@danroth27 
I have the following in [Working with Forms](https://github.com/aspnet/Docs/issues/1027) - but I'd rather move the explanation to this doc.

The data annotations applied to the ``Email`` and ``Password`` properties generated meta data on the model in the form of `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__ ``data-val-*`` attributes (see :doc:`/mvc/models/validation`). (``data-val-`` specifies data validation.) The Input Tag Helper consumes the model metadata and produces HTML5-compatible attributes describing the validators to attach to the input fields. This provides unobtrusive HTML5 and `jQuery <https://jquery.com/>`__ validation. The unobtrusive attributes have the format ``data-val-rule="Error Message"``, where rule is the validation rule (such as required, email format, minimum string length, and maximum string length). If an error message is provided in the attribute, it is displayed as the value for the ``data-val-rule`` attribute. Although the HTML5 ``data-val-*`` attributes were originally targeted for jQuery validation, all the major browsers will generate client some side errors without jQuery (for example, on ``type="email"``).